source/f5_transportserver.go

```diff
@@ -0,0 +1,213 @@
+
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+var f5TransportServerGVR = schema.GroupVersionResource{
+	Group:    "cis.f5.com",
+	Version:  "v1",
+	Resource: "transportservers",
+}
+
+// transportServerSource is an implementation of Source for F5 TransportServer objects.
+type f5TransportServerSource struct {
+	dynamicKubeClient       dynamic.Interface
+	transportServerInformer informers.GenericInformer
+	kubeClient              kubernetes.Interface
+	annotationFilter        string
+	namespace               string
+	unstructuredConverter   *unstructuredConverter
+}
+
+func NewF5TransportServerSource(
+	ctx context.Context,
+	dynamicKubeClient dynamic.Interface,
+	kubeClient kubernetes.Interface,
+	namespace string,
+	annotationFilter string,
+) (Source, error) {
+	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicKubeClient, 0, namespace, nil)
+	transportServerInformer := informerFactory.ForResource(f5TransportServerGVR)
+
+	transportServerInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+			},
+		},
+	)
+
+	informerFactory.Start(ctx.Done())
+
+	// wait for the local cache to be populated.
+	if err := waitForDynamicCacheSync(context.Background(), informerFactory); err != nil {
+		return nil, err
+	}
+
+	uc, err := newTSUnstructuredConverter()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to setup unstructured converter")
+	}
+
+	return &f5TransportServerSource{
+		dynamicKubeClient:       dynamicKubeClient,
+		transportServerInformer: transportServerInformer,
+		kubeClient:              kubeClient,
+		namespace:               namespace,
+		annotationFilter:        annotationFilter,
+		unstructuredConverter:   uc,
+	}, nil
+}
+
+// Endpoints returns endpoint objects for each host-target combination that should be processed.
+// Retrieves all TransportServers in the source's namespace(s).
+func (ts *f5TransportServerSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error) {
+	transportServerObjects, err := ts.transportServerInformer.Lister().ByNamespace(ts.namespace).List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	var transportServers []*f5.TransportServer
+	for _, tsObj := range transportServerObjects {
+		unstructuredHost, ok := tsObj.(*unstructured.Unstructured)
+		if !ok {
+			return nil, errors.New("could not convert")
+		}
+
+		transportServer := &f5.TransportServer{}
+		err := ts.unstructuredConverter.scheme.Convert(unstructuredHost, transportServer, nil)
+		if err != nil {
+			return nil, err
+		}
+		transportServers = append(transportServers, transportServer)
+	}
+
+	transportServers, err = ts.filterByAnnotations(transportServers)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to filter TransportServers")
+	}
+
+	endpoints, err := ts.endpointsFromTransportServers(transportServers)
+	if err != nil {
+		return nil, err
+	}
+
+	// Sort endpoints
```

Comment:
Would you please explain in this comment why the endpoints are sorted ?

Reply:
Don't think it's actually needed so removed it.